src/classes/WordPress.php

```diff
@@ -44,7 +46,8 @@ public function __construct() {
 			add_filter( 'file_mod_allowed', array( $this, 'read_only_filesystem' ), PHP_INT_MAX, 2 );
 			add_filter( 'pre_reschedule_event', array( $this, 'delay_cronjobs' ), PHP_INT_MAX, 2 );
 			add_action( 'admin_enqueue_scripts', array( $this, 'admin_scripts' ) );
-			add_action( 'admin_init', array( &$this, 'check_block_request' ) );
+			add_action( 'admin_init', array( $this, 'check_block_request' ) );
+			add_filter( 'site_url', array( $this, 'staging_url_override' ), PHP_INT_MAX, 4 );
```

Comment:
site_url is already set in wp-config for all deploys. Also, check out https://github.com/azureappserviceoss/wordpress-azure/issues/30 which uses a different approach.